packages/medusa/src/utils/define-routes-config.ts

```diff
@@ -32,8 +32,16 @@ export function defineRoutesConfig<
       next: MedusaNextFunction
     ) => any)[]
   }
->(routes: Route[]): MiddlewaresConfig {
+>(
+  config:
+    | Route[]
+    | { routes?: Route[]; errorHandler?: MiddlewaresConfig["errorHandler"] }
```

Comment:
thought(non-blocking): I wonder if we should rethink the `errorHandler`, so we don't have to jam it into the API like this.

IMO, the `defineRoutesConfig` would ideally only accept `Route[]`.

The `errorHandler` is used to either:
- Opt-out of our default error handler -> `errorHandler: false`
- Inject a custom `errorHandler` -> `errorHandler: () => { ... }`

Both features are not frequently requested, so perhaps we should optimize for a cleaner API. Perhaps, it can be part of a second argument to the function?

What do you think?

Reply:
On second thought, I am not sure if a second `options` (or similar) argument is that much better. Would love to hear what you think about this!

Reply:
IMO it is fine for now, you can either pass routes only or an object that allows you to provide an error handler

Reply:
Is this a global error handler? If so, then we might decouple it from this `defineRoutesConfig` helper. And maybe export it as a standalone value. Something like:

```ts
export const routesConfig = defineRoutesConfig()
export const errorHandler = defineErrorHandler()
```

Not 100% confident with the above suggestion. But, if we will need a few more config values for the API server, then it might make sense to have a server config file for these sort of things.

Reply:
I agree with Oli that we should not have two different signatures for `defineRoutesConfig`. I think passing a second `options` argument is a good approach, since it won't be frequently used anyway.

Reply:
@shahednasser I am thinking more from a fundamental point of view. Is assigning an `errorHandler` a route specific config? 

Since, we have the freedom to rethink/break APIs right now. Maybe we can give the `errorHandler` its own place :)

Reply:
Fine with the `defineErrorHandler` approach too, my only thinking is that as Oli mentioned this feature isn't requested as much.

Reply:
Alright. Let's see if @olivermrbl has more to add here!